x/pki/handler_test.go

```diff
@@ -135,1703 +135,1983 @@ func Setup(t *testing.T) *TestSetup {
-			accAddress.String(), testconstants.RootSubject, testconstants.RootSubjectKeyID, testconstants.Info)
-		_, err := setup.Handler(setup.Ctx, proposeRevokeX509RootCert)
-		require.Error(t, err)
-		require.True(t, sdkerrors.ErrUnauthorized.Is(err))
-	}
-}
+	// propose x509 root certificate by `ownerTrustee`
+	_, err := ProposeX509RootCert(setup, ownerTrustee, options.pemCert, options.info, options.vid)
+	require.NoError(setup.T, err)
 
-func TestHandler_ProposeRevokeX509RootCert_CertificateDoesNotExist(t *testing.T) {
-	setup := Setup(t)
+	// approve x509 root certificate by another trustee
+	_, err = ApproveX509RootCert(setup, setup.Trustee2, options.subject, options.subjectKeyID, options.info)
+	require.NoError(setup.T, err)
 
-	// propose revocation of not existing certificate
-	proposeRevokeX509RootCert := types.NewMsgProposeRevokeX509RootCert(
-		setup.Trustee1.String(), testconstants.RootSubject, testconstants.RootSubjectKeyID, testconstants.Info)
-	_, err := setup.Handler(setup.Ctx, proposeRevokeX509RootCert)
-	require.Error(t, err)
-	require.True(t, pkitypes.ErrCertificateDoesNotExist.Is(err))
+	// check that root certificate has been approved
+	approvedCertificate, err := querySingleApprovedCertificate(
+		setup, options.subject, options.subjectKeyID)
+	require.NoError(setup.T, err)
+	require.NotNil(setup.T, approvedCertificate)
 }
 
-func TestHandler_ProposeRevokeX509RootCert_ForProposedCertificate(t *testing.T) {
-	setup := Setup(t)
-
-	// propose x509 root certificate
-	proposeAddX509RootCert := types.NewMsgProposeAddX509RootCert(setup.Trustee1.String(), testconstants.RootCertPem, testconstants.Info, testconstants.Vid)
-	_, err := setup.Handler(setup.Ctx, proposeAddX509RootCert)
-	require.NoError(t, err)
-
-	// check that proposed certificate is present
-	proposedCertificate, _ := queryProposedCertificate(setup, testconstants.RootSubject, testconstants.RootSubjectKeyID)
-	require.NotNil(t, proposedCertificate)
-
-	// propose revocation of proposed root certificate
-	proposeRevokeX509RootCert := types.NewMsgProposeRevokeX509RootCert(
-		setup.Trustee1.String(), testconstants.RootSubject, testconstants.RootSubjectKeyID, testconstants.Info)
-	_, err = setup.Handler(setup.Ctx, proposeRevokeX509RootCert)
-	require.Error(t, err)
-	require.True(t, pkitypes.ErrCertificateDoesNotExist.Is(err))
-}
+func TestHandler(t *testing.T) {
+	//setup := Setup(t)
+
+	t.Run("ProposeAddX509RootCert", func(t *testing.T) {
+
+		t.Run("ByNotTrustee", func(t *testing.T) {
+			setup := Setup(t)
+
+			for _, role := range []dclauthtypes.AccountRole{
+				dclauthtypes.Vendor,
+				dclauthtypes.CertificationCenter,
+				dclauthtypes.NodeAdmin,
+			} {
+				accAddress := GenerateAccAddress()
+				setup.AddAccount(accAddress, []dclauthtypes.AccountRole{role}, 1)
+
+				// propose x509 root certificate
+				proposeAddX509RootCert := types.NewMsgProposeAddX509RootCert(accAddress.String(), testconstants.RootCertPem, testconstants.Info, testconstants.Vid)
```

Comment:
Why don't we use proposeX509RootCert function here?

Reply:
Didn't notice, fixed it